src/test/java/io/xpring/ilp/DefaultIlpClientTest.java

```diff
@@ -0,0 +1,331 @@
+package io.xpring.ilp;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.AdditionalAnswers.delegatesTo;
+import static org.mockito.Mockito.mock;
+
+import org.interledger.spsp.server.grpc.AccountServiceGrpc;
+import org.interledger.spsp.server.grpc.BalanceServiceGrpc;
+import org.interledger.spsp.server.grpc.CreateAccountRequest;
+import org.interledger.spsp.server.grpc.CreateAccountResponse;
+import org.interledger.spsp.server.grpc.GetAccountRequest;
+import org.interledger.spsp.server.grpc.GetAccountResponse;
+import org.interledger.spsp.server.grpc.GetBalanceRequest;
+import org.interledger.spsp.server.grpc.GetBalanceResponse;
+import org.interledger.spsp.server.grpc.IlpOverHttpServiceGrpc;
+import org.interledger.spsp.server.grpc.SendPaymentRequest;
+import org.interledger.spsp.server.grpc.SendPaymentResponse;
+
+import io.grpc.ManagedChannel;
+import io.grpc.inprocess.InProcessChannelBuilder;
+import io.grpc.inprocess.InProcessServerBuilder;
+import io.grpc.stub.StreamObserver;
+import io.grpc.testing.GrpcCleanupRule;
+import io.xpring.ilp.DefaultIlpClient;
+import io.xpring.xrpl.XpringException;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Represents the result of a gRPC network call for an object of type T or an error.
+ */
+class GRPCResult<T> {
```

Comment:
If you like this class, we should just refactor this to some place common so our tests can use it.